File: app/src/main/java/net/ballmerlabs/uscatterbrain/db/entities/Identity.java
package net.ballmerlabs.uscatterbrain.db.entities;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class Identity {
    @Embedded
    public KeylessIdentity identity;

    @Relation(
            parentColumn = "identityID",
            entityColumn = "identityFK"
    )
    public List<Keys> keys;
}

File: app/src/main/java/net/ballmerlabs/uscatterbrain/db/entities/Identity.kt
package net.ballmerlabs.uscatterbrain.db.entities

import androidx.room.Embedded
import androidx.room.Relation
import java.util.*

/**
 * class with identity and corresponding relations.
 * represents a one-to-many relationship between KeylessIdentity
 * and multiple keys and ACLs
 */
data class Identity(
    @Embedded
    var identity: KeylessIdentity,

    @Relation(parentColumn = "identityID", entityColumn = "identityFK")
    var keys: List<Keys>,

    @Relation(parentColumn = "identityID", entityColumn = "identityFK")
    var clientACL: List<ClientApp>? = null
)


/**
 * internal class for looking up identity by fingerprint
 * in Room
 */
data class JustFingerprint(
        val fingerprint: UUID
)